decrypt-aspera/tests/checker.nf

```diff
@@ -0,0 +1,131 @@
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+
+  Authors:
+    Edmund Su
+*/
+
+/*
+ This is an auto-generated checker workflow to test the generated main template workflow, it's
+ meant to illustrate how testing works. Please update to suit your own needs.
+*/
+
+/********************************************************************/
+/* this block is auto-generated based on info from pkg.json where   */
+/* changes can be made if needed, do NOT modify this block manually */
+nextflow.enable.dsl = 2
+version = '0.1.0'  // package version
+
+container = [
+    'ghrc.io': 'ghrc.io/icgc-argo/argo-data-submission.decrypt-aspera'
+]
+default_container_registry = 'ghrc.io'
+/********************************************************************/
+
+// universal params
+params.container_registry = ""
+params.container_version = ""
+params.container = ""
+
+params.cpus = 1
+params.mem = 1  // GB
+params.publish_dir = ""  // set to empty string will disable publishDir
+
+// tool specific parmas go here, add / change as needed
+params.input_file = ""
+params.expected_output = "./expected/mystery_contents.txt"
+
+process decryptAspera {
+  container "${params.container ?: container[params.container_registry ?: default_container_registry]}:${params.container_version ?: version}"
+  publishDir "${params.publish_dir}/${task.process.replaceAll(':', '_')}", mode: "copy", enabled: params.publish_dir
+  errorStrategy 'terminate'
+  cpus params.cpus
+  memory "${params.mem} GB"
+
+  input:  // input, make update as needed
+    path file
+    path c4gh_secret_key
+  output:  // output, make update as needed
+    path "*.md5", emit: md5_file
+    path "*.{bam,cram,fastq.gz,fq.gz,txt}", emit: output_files
+
+  script:
+    // add and initialize variables here as needed
+    """
+    export C4GH_SECRET_KEY=${c4gh_secret_key}
+    export C4GH_PASSPHRASE='' 
+    python3.6 /tools/main.py \\
+      -f ${file} \\
+      > decrypt.log 2>&1
+
+    """
+}
+
+process file_smart_diff {
+  container "${params.container ?: container[params.container_registry ?: default_container_registry]}:${params.container_version ?: version}"
+
+  input:
+    path output_file
+    path expected_file
+
+  output:
+    stdout()
+
+  script:
+    """
+    # Note: this is only for demo purpose, please write your own 'diff' according to your own needs.
+    # in this example, we need to remove date field before comparison eg, <div id="header_filename">Tue 19 Jan 2021<br/>test_rg_3.bam</div>
+    # sed -e 's#"header_filename">.*<br/>test_rg_3.bam#"header_filename"><br/>test_rg_3.bam</div>#'
+
+    cat ${output_file} ${expected_file} \
+      && ( echo "Test PASSED" && exit 0 ) || ( echo "Test FAILED, output file mismatch." && exit 1 )
+    """
+}
+
+
+workflow checker {
+  take:
+    input_file
+    c4gh_secret_key
+    expected_output
+
+  main:
+    decryptAspera(
+    input_file,
+    params.c4gh_secret_key
+   )
```

Comment:
I need to change to the following to get it work:
```
file(input_file),
file(c4gh_secret_key)
```

Reply:
Added fixed
https://github.com/icgc-argo/argo-data-submission/blob/db4eda6d6a8adb91ed22537142c7f4f4aa70310b/decrypt-aspera/tests/checker.nf#L89-L92